content/en/developers/authorization/oauth2_in_datadog.md

```diff
@@ -0,0 +1,107 @@
+
+   In order to build this URL, use the `site` query parameter that is provided on the redirect to your `redirect_uri`. The format of the domain should match the following format. You can append the `token` path to this constructed URL: `https://{<custom_subdomain>,app,us3,us5}.{datadoghq,ddog-gov}.{com,eu}`.
+
+6. Upon success, you receive your `access_token` and `refresh_token` in the response body. Your application should display a confirmation page with the following message: `You may now close this tab`.
+
+7. Use the `access_token` to make calls to Datadog API endpoints by sending it in as a part of the authorization header of your request: ```headers = {"Authorization": "Bearer {}".format(access_token)}```.
+
+8. Call the [API Key Creation endpoint][7] to generate an API key that allows you to send data on behalf of Datadog users.
+
+   If the `API_KEYS_WRITE` scope has not been added to your client, this step fails. This endpoint generates an API Key that is only shown once. **Store this value in a secure database or location**. 
+
+For more information about client creation and publishing, see the [OAuth for Datadog Integrations][5].
+
+### Initiate authorization from a third-party location 
+
+You can start the authorization process in Datadog by clicking **Connect Accounts** in the integration tile, or from the integration's external website. For example, if there's an integration configuration page on your website that Datadog users need to use, you can give users the option to start the authorization process from there.
+
+When kicking off authorization from a third-party location—anywhere outside of the Datadog integration tile—you need to account for the [Datadog site][17] (such as EU, US1, US3, or US5) when routing users through the authorization flow and building out the URL for the `authorization` and `token` endpoints. 
+
+To ensure that users are authorizing in the correct site, always direct them to the US1 Datadog site (`app.datadoghq.com`), and from there, they can select their region. After the site is selected, ensure that all followup API calls use the selected site. 
+
+## Authorization code grant flow with PKCE
+
+While the OAuth2 protocol supports several grant flows, the [authorization code grant flow][8] [with PKCE](#pkce) is the recommended grant type for long-running applications where a user grants explicit consent once and client credentials can be securely stored. 
+
+This grant type allows applications to securely obtain a unique authorization code and exchange it for an access token that enables them to make requests to Datadog APIs. The authorization code grant flow consists of three steps:
+
+1. The application [requests authorization from a user][6] to access a set of Datadog scopes.
+2. When a user clicks **Authorize**, the application [obtains a unique authorization code][12].
+3. The application [exchanges the authorization code for an access token][10], which is used to access Datadog APIs.
```

Comment:
We should link directly to the token endpoint here instead: 

https://docs-staging.datadoghq.com/anawishnoff/authorization-developer-docs/developers/authorization/oauth2_endpoints/?tab=tokenendpoints#exchange-authorization-code-for-access-token)